R/step_smotenc.R

```diff
@@ -0,0 +1,210 @@
+#' baked <- up_rec %>%
+#'   bake(new_data = credit_data0) %>%
+#'   count(Home, name = "baked")
+#' baked
+#'
+#' # Note that if the original data contained more rows than the
+#' # target n (= ratio * majority_n), the data are left alone:
+#' orig %>%
+#'   left_join(training, by = "Home") %>%
+#'   left_join(baked, by = "Home")
+step_smotenc <-
+  function(recipe, ..., role = NA, trained = FALSE,
+           column = NULL, over_ratio = 1, neighbors = 5,
+           skip = TRUE, seed = sample.int(10^5, 1), id = rand_id("smotenc")) {
+    add_step(
+      recipe,
+      step_smotenc_new(
+        terms = enquos(...),
+        role = role,
+        trained = trained,
+        column = column,
+        over_ratio = over_ratio,
+        neighbors = neighbors,
+        predictors = NULL,
+        skip = skip,
+        seed = seed,
+        id = id
+      )
+    )
+  }
+
+step_smotenc_new <-
+  function(terms, role, trained, column, over_ratio, neighbors, predictors,
+           skip, seed, id) {
+    step(
+      subclass = "smotenc",
+      terms = terms,
+      role = role,
+      trained = trained,
+      column = column,
+      over_ratio = over_ratio,
+      neighbors = neighbors,
+      predictors = predictors,
+      skip = skip,
+      id = id,
+      seed = seed,
+      id = id
+    )
+  }
+
+#' @export
+prep.step_smotenc <- function(x, training, info = NULL, ...) {
+  col_name <- recipes_eval_select(x$terms, training, info)
+  if (length(col_name) > 1) {
+    rlang::abort("The selector should select at most a single variable")
+  }
+
+  if (length(col_name) == 1) {
+    check_column_factor(training, col_name)
+  }
+
+  predictors <- setdiff(info$variable[info$role == "predictor"], col_name)
+  check_type(training[, predictors], TRUE)
+  check_na(select(training, all_of(c(col_name, predictors))), "step_smotenc")
+
+  step_smotenc_new(
+    terms = x$terms,
+    role = x$role,
+    trained = TRUE,
+    column = col_name,
+    over_ratio = x$over_ratio,
+    neighbors = x$neighbors,
+    predictors = predictors,
+    skip = x$skip,
+    seed = x$seed,
+    id = x$id
+  )
+}
+
+#' @export
+bake.step_smotenc <- function(object, new_data, ...) {
+  if (length(object$column) == 0L) {
+    # Empty selection
+    return(new_data)
+  }
+
+  new_data <- as.data.frame(new_data)
+
+  predictor_data <- new_data[, unique(c(object$predictors, object$column))]
+
+  # smotenc with seed for reproducibility
+  with_seed(
+    seed = object$seed,
+    code = {
+      synthetic_data <- smotenc_impl(
+        predictor_data,
+        object$column,
+        k = object$neighbors,
+        over_ratio = object$over_ratio
+      )
+    }
+  )
+  new_data <- na_splice(new_data, synthetic_data, object)
+
+  as_tibble(new_data)
+}
+
+#' @export
+print.step_smotenc <-
+  function(x, width = max(20, options()$width - 26), ...) {
+    cat("SMOTE based on ", sep = "")
+    printer(x$column, x$terms, x$trained, width = width)
```

Comment:
This is a very recent change, you change this to follow this format:
https://github.com/tidymodels/themis/blob/8698526bda3381b53ccf120057cecc8d173bb5ab/R/step_smote.R#L193-L199

Also make sure that the title for this step says SMOTENC instead of SMOTE